#### if...elif..else
#### nested..if


- Syntax

if(condition):
    statements
elif(condition):
    statements
elif(condition):
    statments
    
----
----

else:
    statements

In [2]:
## input: 5
## Output: Five
## input: 0
## Output: Zero

n = int(input('enter number: '))
if(n == 0):
    print('Zero')
elif(n == 1):
    print('One')
elif(n == 2):
    print('Two')
elif(n == 3):
    print('Three')
elif(n == 4):
    print('Four')
elif(n == 5):
    print('Five')
elif(n == 6):
    print('Six')
elif(n == 7):
    print('Seven')
elif(n == 8):
    print('Eight')
elif(n == 9):
    print('Nine')
else:
    print('Invalid')

enter number: 89
Invalid


In [4]:
## Mathematical operations
## input: + ,4,5
## output: 9
## input: - 
## output: -1

a = int(input('enter a number: '))
b = int(input('enter b number: '))
op = input('enter operator: ')
if(op == '+'):
    print(a+b)
elif(op == '-'):
    print(a-b)
elif(op == '*'):
    print(a*b)
elif(op == '/'):
    print(a/b)
elif(op == '%'):
    print(a%b)
else:
    print('Invalid operator')

enter a number: 9
enter b number: 6
enter operator: ;
Invalid operator


-- nested..if Syntax

if(condition):
    if(condition):
        statements
    else:
        statements


In [8]:
### 1--150
## pw: 12345

### u_id
    ### enter pwd: == 12345
        ### Succussfully completed
    ### wrong pwd
### Invalid user_id

u_id = int(input('enter user id: '))
if(u_id>=1 and u_id<=150):
    pwd = int(input('enter pwd: '))
    if(pwd == 12345):
        print('Successfully completed')
    else:
        print('Wrong password')
else:
    print('Invalid user_id')
    
        
    

enter user id: 789
Invalid user_id


## Iterations(Loops)


- Repeat the sequence of code until a specific condition false


- for
- while
- nested for
- nested while

In [11]:
print('Apple')
print('Apple')
print('Apple')
print('Apple')
print("Apple")

Apple
Apple
Apple
Apple
Apple


- Syntax of for loop

for variable in range(start,end,step):
    statements/logic

In [12]:
for i in range(1,10,2):
    print(i)

1
3
5
7
9


In [ ]:
i = 1
i + 2
1+2 = 3

In [13]:
for i in range(5):
    print(i)

0
1
2
3
4


In [14]:
for i in range(1,10):
    print(i)

1
2
3
4
5
6
7
8
9


In [21]:
for i in range(1,10):
    print(i,end= ' ')

1 2 3 4 5 6 7 8 9 

In [28]:
for i in range(10,0,-1):
    print(i,end=' ')

10 9 8 7 6 5 4 3 2 1 

In [32]:
## Sum of n natural numbers

## input: 10
## output: Sum of 10 number is: 55

n = int(input())
s = 0
for i in range(1,n+1):
    s = s + i ## s += i
print('Sum of',n,'numbers is: ',s)


10
Sum of 10 numbers is:  55


In [33]:
sr = int(input())
er = int(input())
s = 0
for i in range(sr,er+1):
    s += i
print(s)

23
30
212


In [34]:
## Table 
## 5
## 5 * 1 = 5
## 5 * 2 = 10
# --
# --
## 5 * 10 = 50

t = int(input())
for i in range(1,11):
    print(t,'*',i,'=',t*i)

5
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


In [40]:
### print alphabets
## a to z  97 -- 122
## A to Z  65 -- 90
print('Lower case letters are: ')
for i in range(97,123):
    print(chr(i),end=' ')
print('\nUpper case letters are: ')
for j in range(65,91):
    print(chr(j),end=' ')

Lower case letters are: 
a b c d e f g h i j k l m n o p q r s t u v w x y z 
Upper case letters are: 
A B C D E F G H I J K L M N O P Q R S T U V W X Y Z 

In [38]:
print(ord('a'))
print(ord('z'))
print(ord('A'))
print(ord('Z'))
print(chr(99))

97
122
65
90
c


In [47]:
## 1 -- 10
## Even numbers are:
## 2 4 6 8 10
## Odd number are:
## 1 3 5 7 9

sr = int(input())
er = int(input())
print('Even numbers are: ')
for i in range(sr,er+1):
    if(i%2 == 0):
        print(i,end=' ')
print('\nOdd numbers are')
for j in range(sr,er+1):
    if(j%2 == 1): ## j%2 != 0
        print(j,end=' ')

67
156
Even numbers are: 
68 70 72 74 76 78 80 82 84 86 88 90 92 94 96 98 100 102 104 106 108 110 112 114 116 118 120 122 124 126 128 130 132 134 136 138 140 142 144 146 148 150 152 154 156 
Odd numbers are
67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99 101 103 105 107 109 111 113 115 117 119 121 123 125 127 129 131 133 135 137 139 141 143 145 147 149 151 153 155 

In [50]:
## Prime number

n = int(input())
fc = 0
for i in range(1,n+1):
    if(n%i == 0):
        fc = fc + 1  ## fc += 1
if(fc == 2):
    print('Prime Number')
else:
    print("Not prime number")
    
        

9
Not prime number


### Task

1. Positive or Negative
2. Check the given number is divisible by 5 and as well as 11
3. Days printing
4. 
    - Child (0--12 years)
    - Teenager (13 -- 18 years)
    - Adult (19 --59 years)
    - Senior adult (60  and above)
5. Check the given input is Digit or Alphabet  or Special symbol



In [6]:
## Positive or Negative

n = int(input('enter number: '))
if(n>0):
    print('Positive number')
elif(n<0):
    print('Negative number')
else:
    print('Zero')

enter number: 56
Positive number


In [8]:
## Check the given number is divisible by 5 and as well as 11

num = int(input('enter number: '))
if((num%5==0) and (num%11==0)):
    print(num)


enter number: 55
55


In [10]:
## Days printing

t = int(input())
if(t==1):
    print('Sunday')
elif(t==2):
    print('Monday')
elif(t==3):
    print('Tuesday')
elif(t==4):
    print('Wednesday')
elif(t==5):
    print('Thursday')
elif(t==6):
    print('Friday')
elif(t==7):
    print('Saturday')
else:
    print('Wrong choice')

89
Wrong choice


In [15]:
#  - Child (0--12 years)
#     - Teenager (13 -- 18 years)
#     - Adult (19 --59 years)
#     - Senior adult (60  and above)

age = int(input())
if(age>0):
    if(age>0 and age<=12):
        print('Child')
    elif(age>=13 and age<=18):
        print('Teenager')
    elif(age>=19 and age<=59):
        print('Adult')
    else:
        print('Senior adult')
else:
    print('Invalid')

-9
Invalid


In [19]:
# Check the given input is Digit or Alphabet  or Special symbol
## 0---9 -->Digit
## a--z or A--Z --> Alphabets
## special symbol

ch = input('enter input: ')
if(ch>='0' and ch<='9'):
    print('Digit')
elif((ch>='a' and ch<='z') or (ch>='A' and ch<='Z')):
    print('Alphabet')
else:
    print('Special Symbol')
    


enter input: 99
Special Symbol


In [25]:
## factorial of number

t = int(input('enter number: '))
f = 1
if(t == 0):
    print(f)
else:
    for i in range(t,0,-1):
        f = f*i
    print(f)

enter number: 6
720


In [22]:

t = int(input('enter number: '))
f = 1
for i in range(1,t+1):
    f = f*i
print(f)

enter number: 5
120


## while

- Used to repeat a block of code or sequence as long as the condition false
- Syntax:
    intialization
    while(condition):
        statements/logic
        incre/decre
- Syntax
- execution speed is faster than for loop
- number of iterations unknown


In [26]:
## printing n natural 
n = int(input('enter number: '))
i = 1  ## intialization
while(i<=n): ## 1<=10 2<=10 3<=10 --- 10<=10
    print(i,end=' ')
    i += 1 ## i = i+1
    # i -= 1 ## i = i-1

    

enter number: 10
1 2 3 4 5 6 7 8 9 10 

In [1]:
n = int(input())
i = 10
while(i>=n):
    print(i,end=' ')
    i = i -1 ## i -= 1

1
10 9 8 7 6 5 4 3 2 1 

In [7]:
## Number of digits in the given number
## input: 5678
## output: 4
## 5+6+7+8 = 26

n = int(input())
d_c = 0
d_sum = 0
while(n!=0):
    rem = n%10
    n = n//10
    d_c = d_c+1
    d_sum = d_sum + rem
print('Digit count is: ',d_c)
print('Digits sum is: ',d_sum)

    
    

5678
Digit count is:  4
Digits sum is:  26


In [2]:
n = 5678
n//10

567

In [3]:
n = 567
n//10

56

In [4]:
n = 56
n//10

5

In [9]:
### reverse of the given number
## input: 123
## output: 321


r = int(input())
rev = 0
while(r!=0):
    rem = r%10
    rev = rev*10+rem
    r = r//10
print(rev)


345
543


In [10]:
12//10

1

## Tasks

- N natural numbers sum using while loop
- check the given number is perfect or not by using for loop
- check the given number is paindrom or not by using while loop
    - input: 121 -- rev = 121
    - output: palindrom
- check the given number is Armstrong or not
    - input: 153  1^3+5^3+3^3 = 1+125+27 = 153
    - outout: Armstrong

In [51]:
r = int(input('enter number: '))
if(r>=1 and r<=100):
    print('existed')
else:
    print('not existed')

enter number: 6
existed
